tests/libvmi/lifecycle.go

```diff
@@ -1,46 +1,121 @@
 package libvmi
 
 import (
+	"fmt"
 	"time"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
 
 	v1 "kubevirt.io/client-go/api/v1"
 	"kubevirt.io/client-go/kubecli"
 
 	"kubevirt.io/kubevirt/tests"
 )
 
-func SetupVMI(vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
+type VMIPool map[string]*v1.VirtualMachineInstance
+
+func NewVMIPool() VMIPool {
+	return map[string]*v1.VirtualMachineInstance{}
+}
+
+func (vmis VMIPool) Setup(loggedInExpecterFactory tests.VMIExpecterFactory) {
 	virtClient, err := kubecli.GetKubevirtClient()
 	Expect(err).NotTo(HaveOccurred(), "Should initialize KubeVirt client")
 
-	By("Creating the VMI")
-	vmi, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
-	Expect(err).NotTo(HaveOccurred(), "VMI should be successfully created")
+	vmis.create(loggedInExpecterFactory, virtClient)
+	vmis.waitUntilReady(loggedInExpecterFactory, virtClient)
+}
 
-	By("Waiting until the VMI gets ready")
-	vmi = tests.WaitUntilVMIReady(vmi, tests.LoggedInCirrosExpecter)
+func (vmis VMIPool) create(loggedInExpecterFactory tests.VMIExpecterFactory, virtClient kubecli.KubevirtClient) {
+	for name, vmi := range vmis {
+		By(fmt.Sprintf("Creating VMI %q", name))
+		vmi, err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(vmi)
+		Expect(err).NotTo(HaveOccurred(), "VMI %q should be successfully created", name)
+		vmis[name] = vmi
+	}
+}
 
-	return vmi
+func (vmis VMIPool) waitUntilReady(loggedInExpecterFactory tests.VMIExpecterFactory, virtClient kubecli.KubevirtClient) {
+	for name, vmi := range vmis {
+		By(fmt.Sprintf("Waiting until VMI %q gets ready", name))
+		vmis[name] = tests.WaitUntilVMIReady(vmi, loggedInExpecterFactory)
+	}
 }
 
-func CleanupVMI(vmi *v1.VirtualMachineInstance) {
-	if vmi != nil {
-		virtClient, err := kubecli.GetKubevirtClient()
-		Expect(err).NotTo(HaveOccurred(), "Should initialize KubeVirt client")
+func (vmis VMIPool) Cleanup() {
+	virtClient, err := kubecli.GetKubevirtClient()
+	Expect(err).NotTo(HaveOccurred(), "Should initialize KubeVirt client")
 
-		By("Deleting the VMI")
-		Expect(virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.GetName(), &metav1.DeleteOptions{})).To(Succeed())
+	errs := []error{}
+	errs = append(errs, vmis.delete(virtClient)...)
+	if err = vmis.waitUntilDisposed(virtClient); err != nil {
+		errs = append(errs, err)
+	}
+	Expect(errs).To(BeEmpty(), "Cleanup of all VMIs should succeed")
+}
+
+func (vmis VMIPool) delete(virtClient kubecli.KubevirtClient) []error {
+	errs := []error{}
+
+	for name, vmi := range vmis {
+		if vmi == nil {
+			continue
+		}
+
+		By(fmt.Sprintf("Deleting VMI %q", name))
+		err := virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(vmi.GetName(), &metav1.DeleteOptions{})
```

Comment:
food for thought
use vmi.Namespace instead of tests.NamespaceTestDefault

(all over in case there are more places)
better to be self contained.